aiida/cmdline/params/options/main.py

```diff
@@ -110,8 +111,10 @@ def set_log_level(_ctx, _param, value):
     log.CLI_ACTIVE = True
 
     # If the value is ``None``, it means the option was not specified, but we still configure logging for the CLI
+    # However, we skip this when we are in a tab-completion context.
     if value is None:
-        configure_logging()
+        if not _ctx.resilient_parsing:
+            configure_logging()
```

Comment:
I am a bit surprised by this change. This function `set_log_level` is only assigned as the callback of the `VERBOSITY` option. I don't think this is supposed to be called during tab-completion anyway. I just tested this and it indeed doesn't seem to be called during tab-completion. Was this the part of the code that caused the new test to fail? Do you understand why?

Reply:
Good catch, sorry for not being clear, I am also confused, but you can try when you remove it the test fails. But when I test the completion on the actual command line it is not called. Maybe the click function used in the test is not exactly the one that gets called?? Btw: I was testing on BASH, wonder if other shells may behave differently.

Reply:
Perhaps there is some weird interaction with the test suite. Not sure if it is worth deeper investigation, since the change itself seems like an okay thing to do on its own.

Reply:
Figured it out. I tested whether this was being called when actually tab-completing `verdi` by adding a `print` statement. Since that print statement never showed up, I concluded that the function wasn't being called. But that is not true. It _was_ actually called, but during tab-completion, all output to `sys.stdout` is captured and so I didn't see anything. Printing to `sys.stderr` would actually show, or simply raising an exception would confirm the function was being called.